### DISCUSSION TITLE
Understanding differences between Remote and Embedded versions of PyMechancial

### DISCUSSION THREAD

**Discussion by RGPATCHI** [ansys/pymechanical#476] [General]
Hi All, @pmaroneh @koubaa,

I've started working with PyMechanical recently and I would like to check if my understanding regarding the differences  between the two versions (Remote vs Embedded) is correct. 

With Remote:

1. I cannot use cpython commands inside run_python_script_from_file() as Mechanical doesn't support them. 
2. I have to keep passing strings into run_python_script_from_file() function for passing variables from python space to mechanical space.
3. I can open Mechanical in GUI during python script execution
4. I can connect to remote instance

With Embedded: 

1. I can use cpython normally along with mechanical scripting commands. 
2. All python variables are available in Mechanical space as well and no transfer is needed.
3. No GUI available during python script execution 
4. Only Local instance

I'm sure there are more differences, so please let me know if I missed any important difference.

**Comment by dipinknair** [comment]
#511

**Comment by koubaa** [comment]
Explanation is now live: https://mechanical.docs.pyansys.com/version/dev/architecture.html#ref-architecture

Next release, it'll appear in the stable docs
